prince-of-versions/src/main/java/co/infinum/princeofversions/JsonConfigurationParser.java

```diff
@@ -224,8 +224,12 @@ private boolean parseJsonUpdate(JSONObject update, PrinceOfVersionsConfig.Builde
         Iterator<String> it = requirementsJson.keys();
         while (it.hasNext()) {
             String key = it.next();
-            Object value = requirementsJson.get(key);
-            requirements.put(key, String.valueOf(value));
+            Object value = (!requirementsJson.isNull(key)) ? requirementsJson.get(key) : null;
+            if (value == null) {
+                requirements.put(key, null);
```

Comment:
Be careful with this, requirements expect non-null value (as not annotation with Nullable). Be sure to handle that too

Reply:
I was under the impression that the `requirementsProcessor` handles the possible null since it can't parse it in `requirementsProcessor.areRequirementsSatisfied(requirements)` & returns false, thereby showing that the update cannot continue.

Reply:
Correct, it will recover from that case, but that shouldn't be valid case. The try-catch is there just because RequirementsChecker is user implemented code and we want to skip that update if it throws. But that doesn't mean we should pass the null value to it. Null is similar as the key doesn't exist in json at all.